### PR TITLE
[Reader IA] Update Reader strings

### LINKED ISSUE #19961: [Reader IA] Update all “Follow” and “Following” terms to “Subscribe” and “Subscribed”
Context: p1703152769063979-slack-C05N140C8H5

### PR DESCRIPTION
**Pull request by RenanLukas**
Fixes #19961 

Update Reader strings to new terminologies:

- `Site` -> `Blog`
- `Follow` -> `Subscribe`
- `Topic` -> `Tag`

-----

## To Test:

- Launch Jetpack and login
- Do a general smoke test of Reader screens
- Look for any of the following strings:
   - Follow, follows, unfollow, following, etc
   - Site, sites
   - Topic, topics

-----

## Regression Notes

1. Potential unintended areas of impact

    - Reader labels

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing, updated unit tests

3. What automated tests I added (or what prevented me from doing so)

    --

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## UI Changes Testing Checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1636,29 +1636,29 @@
     <string name="reader_filter_cta">Filter</string>
     <string name="reader_filter_by_blog_title">Filter by blog</string>
     <string name="reader_filter_by_tag_title">Filter by tag</string>
-    <string name="reader_filter_empty_sites_list">See the newest posts from sites you follow</string>
-    <string name="reader_filter_empty_tags_list">You can follow posts on a specific subject by adding a topic</string>
-    <string name="reader_filter_self_hosted_empty_sites_list">Log in to WordPress.com to see the latest posts from sites you follow</string>
-    <string name="reader_filter_self_hosted_empty_tagss_list">Log in to WordPress.com to see the latest posts from topics you follow</string>
-    <string name="reader_filter_empty_sites_action">Follow a site</string>
-    <string name="reader_filter_empty_tags_action">Add a topic</string>
+    <string name="reader_filter_empty_blogs_list">See the newest posts from blogs you\'re subscribed to</string>
```
Comment (thomashorta) [inline WordPress/src/main/res/values/strings.xml:1639]: ❓ I see usage of both `’` and `\'` for the apostrophe character in our strings file, do you know if any of them is "wrong"?

I noticed that recent Figma design copies use `’`, so I'm not sure if we should be using that.

File: WordPress/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -2126,8 +2126,8 @@
     <!-- accessibility content desciptions -->
     <string name="reader_btn_remove_filter_content_description">Remove the current filter</string>
     <string name="reader_btn_select_filter_content_description">Select a Site or Tag to filter posts</string>
```
Comment (thomashorta) [inline WordPress/src/main/res/values/strings.xml]: 💡 It looks like this uses "site" and "tag" but at the same time it's an unused resource so maybe we can just get rid of it in this `main` `strings.xml` file?

File: WordPress/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -2224,18 +2224,18 @@
...
     <string name="reader_empty_posts_in_tag_updating">Fetching posts…</string>
-    <string name="reader_empty_posts_in_custom_list">The sites in this list haven\'t posted anything recently</string>
-    <string name="reader_empty_followed_tags_subtitle">Add topics here to find posts about your favorite topics</string>
-    <string name="reader_empty_followed_tags_title">No followed topics</string>
-    <string name="reader_empty_recommended_blogs">No recommended sites</string>
-    <string name="reader_empty_followed_blogs_title">No followed sites</string>
-    <string name="reader_empty_followed_blogs_search_title">No sites matching your search</string>
-    <string name="reader_empty_followed_blogs_description">When you follow sites, you\'ll see their content here</string>
+    <string name="reader_empty_blogs_posts_in_custom_list">The blogs in this list have not posted anything recently</string>
+    <string name="reader_empty_subscribed_tags_subtitle">Add tags here to find posts about your favorite tags</string>
```
Comment (thomashorta) [inline WordPress/src/main/res/values/strings.xml]: 💡 I think the last "topics" in this string was meant as the actual meaning of an interest topic, so the sentence doesn't make much sense now. I would suggest keeping the word "topics" there or maybe change to "interests", wdyt?

vorite tags
```suggestion
    <string name="reader_empty_subscribed_tags_subtitle">Add tags here to find posts about your favorite topics</string>
```

File: WordPress/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -2083,22 +2083,22 @@
...
     <string name="reader_title_photo_viewer">%1$d of %2$d</string>
     <string name="reader_title_comments" translatable="false">@string/comments</string>
-    <string name="reader_title_interests">Follow topics</string>
+    <string name="reader_title_tags">Follow tags</string>
 
     <!-- view pager titles -->
-    <string name="reader_page_followed_tags">Followed topics</string>
-    <string name="reader_page_followed_blogs">Followed sites</string>
+    <string name="reader_page_followed_tags_title">Followed tags</string>
+    <string name="reader_page_followed_blogs_title">Followed blogs</string>
```
Comment (thomashorta) [inline WordPress/src/main/res/values/strings.xml]: ❓  Is it ok to keep "follow"/"followed" in this situation or should it be replaced with "Subscribe"/"Subscribed"?

**Review by thomashorta — COMMENTED** [review]
I added a few questions and suggestions.

Also, while reviewing I noticed we refer to "follow" in SEVERAL places across the app (like stats, settings, people management) so I wonder how confusing using "subscribe" in the Reader context feels to the users. I'm a bit less confident that we should update this nomenclature without changing other places as well or double checking what the web does. 😞

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (RenanLukas) [reply WordPress/src/main/res/values/strings.xml:1639]: It seems like in the past using `'` directly wouldn't work (see [here](https://stackoverflow.com/questions/39424713/how-to-type-apostrophe-character-in-android-strings-xml)). I'm not sure it works properly nowadays, but for sure `\'` works.

File: WordPress/src/main/res/values/strings.xml (hunk 2 above)
Comment (RenanLukas) [reply WordPress/src/main/res/values/strings.xml]: Thanks, removed.

File: WordPress/src/main/res/values/strings.xml (hunk 3 above)
Comment (RenanLukas) [reply WordPress/src/main/res/values/strings.xml]: Makes sense 👍

File: WordPress/src/main/res/values/strings.xml (hunk 4 above)
Comment (RenanLukas) [reply WordPress/src/main/res/values/strings.xml:2097]: Good catch, should be replaced

**Comment by RenanLukas** [comment]
Thanks for the review, @thomashorta 

I've fixed/replied to your comments.

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (thomashorta) [reply WordPress/src/main/res/values/strings.xml:1639]: Just a note that the Unicode character that I sent above and that we use in several strings ([right single quotation mark, U+2019](https://en.wikipedia.org/wiki/Right_single_quotation_mark), or `’`) like [this one](https://github.com/wordpress-mobile/WordPress-Android/blob/a884c102cb5eb161c36797e5214f69e8926edea2/WordPress/src/main/res/values/strings.xml#L4309) is not the same one that you sent ([apostrophe, U+0027](https://en.wikipedia.org/wiki/%27_(disambiguation)), or `'`). 

It looks like the right single quotation mark doesn't need to be escaped (even though there are [some strings](https://github.com/wordpress-mobile/WordPress-Android/blob/a884c102cb5eb161c36797e5214f69e8926edea2/WordPress/src/main/res/values/strings.xml#L3018) in our code that do anyway, so it's probably optional).

It looks like there's some discussion over the internet about which one should be used for apostrophes, but I honestly think any should do. 

I think the problem we have right now is that we use both and we should probably try to use just one for consistency, but no need to change anything in your PR.

**Review by thomashorta — APPROVED** [review]
Thanks for the changes! It looks good to me!

**Pull request merged**